packages/cleanup-plugin/index.ts

```diff
@@ -0,0 +1,86 @@
+import {EditorPlugin, ui, data} from '@wonderlandengine/editor-api';
+import {existsSync} from 'node:fs';
+
+/**
+ * Plugin to cleanup resources with broken links.
+ */
+export default class CleanupPlugin extends EditorPlugin {
+
+    result: Record<string, Array<string>> = {};
+
+    /* The constructor is called when your plugin is loaded */
+    constructor() {
+        super();
+        this.name = 'Project Cleanup Plugin';
+    }
+
+    /* Use this function for drawing UI */
+    draw() {
+        ui.text('Unused Resources');
+        ui.separator();
+
+        if (this.result) {
+            for (let k of Object.keys(this.result)) {
+                ui.text(`Found ${this.result[k].length.toString()} unused ${k}`);
+            }
+
+            ui.separator();
+            if (ui.button('Refresh')) {
+                this.collectResources();
+            }
+            ui.sameLine();
+            if (ui.button('Delete all')) {
+                this.cleanup();
+            }
+        } else {
+            this.collectResources();
+        }
+    }
+
+    LINK_CACHE: Record<string, boolean> = {};
```

Comment:
That doesn't quite work, since then we have no way to know whether we ran `existsSync` for the file yet 🤔 i.e. three states: exists in map, true in map, false in map. Whereas the Set could only store two of them.